src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java

```diff
@@ -753,17 +744,24 @@ public Element getLinkedElement(Element referer, String signature) {
 
     @Override
     public Content specTagOutput(Element holder, List<? extends SpecTree> specTags) {
-        ContentBuilder body = new ContentBuilder();
-        for (SpecTree st : specTags) {
-            appendSeparatorIfNotEmpty(body);
-            body.add(specTagToContent(holder, st));
+        if (specTags.isEmpty()) {
+            return Text.EMPTY;
         }
-        if (body.isEmpty())
-            return body;
+
+        List<Content> links = specTags.stream()
+                .map(st -> specTagToContent(holder, st))
+                .collect(Collectors.toList());
+
+        // Use a different style if any link label is longer than 30 chars or contains commas.
+        boolean hasLongLabels = links.stream().anyMatch(this::isLongOrHasComma);
+        var specList = HtmlTree.UL(hasLongLabels ? HtmlStyle.tagListLong : HtmlStyle.tagList);
+        links.stream()
+                .filter(Predicate.not(Content::isEmpty))
+                .forEach(item -> specList.add(HtmlTree.LI(item)));
```

Comment:
There's a similar not-DRY comment. Perhaps both could be dealt with by extracting the decision on styling into a method?

Reply:
I'd prefer to avoid minor refactoring like this at this stage;  there may be more refactoring in this area in the next release.